omnigibson/action_primitives/curobo.py

```diff
@@ -0,0 +1,590 @@
+            target_pose = inv_robot_pose.view(1, 4, 4) @ target_pose
+            target_pos = target_pose[:, :3, 3]
+            target_quat = T.mat2quat(target_pose[:, :3, :3])
+
+        # Map xyzw -> wxyz quat
+        target_quat = target_quat[:, [3, 0, 1, 2]]
+
+        # Create IK goal
+        ik_goal_batch = lazy.curobo.types.math.Pose(
+            position=self._tensor_args.to_device(target_pos).contiguous(),
+            quaternion=self._tensor_args.to_device(target_quat).contiguous(),
+        )
+
+        # Construct initial state
+        q_pos = th.stack([self.robot.get_joint_positions()] * self.batch_size, axis=0)
+        q_vel = th.stack([self.robot.get_joint_velocities()] * self.batch_size, axis=0)
+        q_eff = th.stack([self.robot.get_joint_efforts()] * self.batch_size, axis=0)
+        sim_js_names = list(self.robot.joints.keys())
+        cu_js_batch = lazy.curobo.types.state.JointState(
+            position=self._tensor_args.to_device(q_pos),
+            velocity=self._tensor_args.to_device(q_vel) * 0.0,
+            # TODO: Ideally this should be nonzero, but curobo fails to compute a solution if so
+            acceleration=self._tensor_args.to_device(q_eff) * 0.0,
+            jerk=self._tensor_args.to_device(q_eff) * 0.0,
+            joint_names=sim_js_names,
+        ).get_ordered_joint_state(self.mg.kinematics.joint_names)
+
+        # Attach object to robot if requested
+        if attached_obj is not None:
+            obj_paths = [geom.prim_path for geom in attached_obj.root_link.collision_meshes.values()]
+            assert len(obj_paths) <= 32, f"Expected obj_paths to be at most 32, got: {len(obj_paths)}"
+            # if len(obj_paths) > 5:
+            #     # Use visual paths instead
+            #     obj_paths = [geom.prim_path for geom in attached_obj.root_link.visual_meshes.values()]
+            self.mg.attach_objects_to_robot(
+                joint_state=cu_js_batch,
+                object_names=obj_paths,
+            )
+
+        # Run batched planning
+        if self.debug:
+            self.mg.store_debug_in_result = True
+
+        result = self.mg.plan_batch(cu_js_batch, ik_goal_batch, plan_cfg)
+
+        if self.debug:
+            from IPython import embed
+
+            embed()
+
+        # Detach attached object if it was attached
+        if attached_obj is not None:
+            self.mg.detach_object_from_robot()
+
+        if return_full_result:
+            return result
+
+        else:
+            succ = th.any(result.success).item()
+            if succ:
+                # Sample random successful path
+                successful_paths = result.get_successful_paths()
+                cmd_plan = successful_paths[np.random.choice(len(successful_paths))]
+                return self.path_to_joint_trajectory(path=cmd_plan, joint_names=sim_js_names)
+
+            else:
+                print("CuRobo plan did not converge to a solution.")
```

Comment:
nit, maybe explicitly  `return None` in the end?

Reply:
removed, now always returns a result